pilot/pkg/security/authz/builder/builder.go

```diff
@@ -303,6 +305,20 @@ func (b Builder) buildHTTP(rules *rbacpb.RBAC, shadowRules *rbacpb.RBAC, provide
 			ConfigType: &hcm.HttpFilter_TypedConfig{TypedConfig: protoconv.MessageToAny(extauthz.http)},
 		},
 	}
+	// By default, HttpConnectionManager buffers 1MiB (1048576B), so we have to adjust its buffer
+	// when ExtAuthz maxRequestBytes is greater than 1MiB. Otherwise, ExtAuthz would not receive more data than this limit.
+	if extauthz.http.GetWithRequestBody().GetMaxRequestBytes() > 1048576 {
+		httpBuffer := &hcm.HttpFilter{
+			Name: wellknown.Buffer,
+			ConfigType: &hcm.HttpFilter_TypedConfig{TypedConfig: protoconv.MessageToAny(&buffer.Buffer{
+				MaxRequestBytes: &wrappers.UInt32Value{Value: extauthz.http.GetWithRequestBody().MaxRequestBytes},
+			})},
+		}
+		// insert buffer between rbac and ext_authz
+		httpFilters = append(httpFilters[:1], httpFilters[0:]...)
+		httpFilters[1] = httpBuffer
```

Comment:
nit: can we just build this in-order to begin with instead of shuffling things around?

Reply:
Do you mean that the buffer should be first?